File: docs/running-parsec-gem5/global_variable_cache/global_variable_cache.cc
#include "global_variable_cache/global_variable_cache.hh"

#include <iostream>

int given_seed_value;

namespace gem5
{

GlobalVariableCache::GlobalVariableCache(const GlobalVariableCacheParams &params) :
    SimObject(params),
    inputSeed(params.inputSeed)
{
    // given_seed_value = params.givenSeed;
    // std::cout << "HENLO" << std::endl;
    std::cout << "instantiated global cache with seed " << inputSeed << std::endl;
    given_seed_value = inputSeed;

}

} // namespace gem5

File: docker-share/gem5/src/global_variable_cache/GlobalVariableCache.py
from m5.params import *
from m5.SimObject import SimObject

class GlobalVariableCache(SimObject):
    type = 'GlobalVariableCache'
    cxx_header = "global_variable_cache/global_variable_cache.hh"
    cxx_class = "gem5::GlobalVariableCache"
    inputSeed = Param.Int(0, "")